### DISCUSSION TITLE
Allowlists/Denylists for models (and Ollama tags)

### DISCUSSION THREAD

**Discussion by secondtruth** [mckaywrigley/chatbot-ui#1422] [Ideas]
I propose introducing an allowlists and denylists feature to control the availability of specific models and tags within Chatbot UI. This would provide admins the possibility to deny or exclusively allow the use of certain models or tags per provider (OpenAI, [Hugging Face](https://github.com/mckaywrigley/chatbot-ui/discussions/1415), Ollama, etc.).

These model lists could be configured through a file, let's say `model-list.json`, in the beginning. This could also be later advanced to a graphical configuration interface, which also complements the vision of a [general admin area](https://github.com/mckaywrigley/chatbot-ui/discussions/1268).

Suggestions and improvements of this proposal are always very appreciated!

## See also

- https://github.com/mckaywrigley/chatbot-ui/discussions/1319

**Comment by spammenotinoz** [comment]
Yes, good idea, some of the models are expensive like GPT4 and Perplexity Online and I want to share this with my family, disabling models is a good idea.
Is there a manual workaround code update to remove certain models?
